data/TaskMemoryDAO.test.js

```diff
@@ -0,0 +1,364 @@
+      const dao = new TaskMemoryDAO();
+
+      chai.assert.isNull(dao.newTask());
+      chai.assert.isNull(dao.newTask(null));
+      chai.assert.isNull(dao.newTask({}));
+      chai.assert.isNull(dao.newTask(12));
+      chai.assert.isNull(dao.newTask(title = 'something', course = 'eo')); // param is not an object
+
+      chai.assert.isNull(dao.newTask({ title: 'some task' }));
+      chai.assert.isNull(dao.newTask({ course: 'course7889' }));
+      chai.assert.isNull(dao.newTask({ title: 'some task', otherProp: 'some other property' }));
+      chai.assert.isNull(dao.newTask({ course: 'course7889', otherProp: 'some other property' }));
+      chai.assert.isNull(dao.newTask({ title: 'some task', otherProp: 'some other property' }));
+    });
+    it('should return a successfully created task', () => {
+      const dao = new TaskMemoryDAO();
+      const task = dao.newTask({ title: 'my task', course: 'CS5000' });
+      chai.assert.isNotNull(task);
+      chai.assert.strictEqual(task.title, 'my task');
+      chai.assert.strictEqual(task.course, 'CS5000');
+    });
+    it('should set default values for non-required properties', () => {
+      const dao = new TaskMemoryDAO();
+      const task = dao.newTask({ title: 'my task', course: 'CS5000' });
+      chai.assert.isNotNull(task);
+      chai.assert.isNumber(task.id);
+      chai.assert.isNull(task.deadline);
+      chai.assert.isFalse(task.completed);
+      chai.assert.isNull(task.scheduledDate);
+    });
+    it('should set all provided properties', () => {
+      const dao = new TaskMemoryDAO();
+      const props = {
+        title: 'my task',
+        course: 'CS5000',
+        deadline: 'Jan 4, 2020',
+        completed: true,
+        scheduledDate: 'Jan 1, 2020',
+      };
+      const task = dao.newTask(props);
+      chai.assert.strictEqual(task.title, 'my task');
+      chai.assert.strictEqual(task.course, 'CS5000');
+      chai.assert.strictEqual(task.deadline, 'Jan 4, 2020');
+      chai.assert.isTrue(task.completed);
+      chai.assert.strictEqual(task.scheduledDate, 'Jan 1, 2020');
+    });
+    it('should add the new task to the tasks array', () => {
+      const dao = new TaskMemoryDAO();
+
+      dao.newTask({ title: 'my task', course: 'CS5000' });
+      chai.assert.lengthOf(dao.tasks, 1);
+      chai.assert.strictEqual(dao.tasks[0].title, 'my task');
+      chai.assert.strictEqual(dao.tasks[0].course, 'CS5000');
+
+      dao.newTask({ title: 'my task2', course: 'MS3000' });
+      chai.assert.lengthOf(dao.tasks, 2);
+      chai.assert.strictEqual(dao.tasks[1].title, 'my task2');
+      chai.assert.strictEqual(dao.tasks[1].course, 'MS3000');
+
+      dao.newTask({ title: 'my task3', course: 'ENGL3000' });
+      chai.assert.lengthOf(dao.tasks, 3);
+      chai.assert.strictEqual(dao.tasks[2].title, 'my task3');
+      chai.assert.strictEqual(dao.tasks[2].course, 'ENGL3000');
+    });
+  });
+  describe('Querying', () => {
+    describe('find()', () => {
```

Comment:
You can make your tests more terse by running a `beforeEach()` to setup for each test

```javascript
 describe('Querying', () => {
    let dao;
    beforeEach(() => {
      dao = new TaskMemoryDAO();
      dao.newTask({ title: 'task1', course: 'CS5000', completed: true });
        dao.newTask({ title: 'task2', course: 'CS4000' });
        dao.newTask({ title: 'task3', course: 'CS5555' });
        dao.newTask({ title: 'task4', course: 'CS5000' });
        dao.newTask({ title: 'task5', course: 'CS5980' });
        dao.newTask({ title: 'task6', course: 'CS3000' });
    });
```